### ISSUE TITLE
CloudGuard API's are returning empty response.

### ISSUE THREAD

**Issue by bigdatasourav** [oracle/oci-go-sdk#317]
CloudGuard API's for Detector receipe, responder receipe, Target and managed list are returning empty response even there is data. Below is one example -- 

```
package main

import (
	"context"
	"fmt"

	"github.com/oracle/oci-go-sdk/v44/cloudguard"
	"github.com/oracle/oci-go-sdk/v44/common"
)

func main() {
	c, err := cloudguard.NewCloudGuardClientWithConfigurationProvider(common.DefaultConfigProvider())
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	tenancyID := "ocid1.tenancy.oc1..*****************************"
	request := cloudguard.ListDetectorRecipesRequest{
		CompartmentId: &tenancyID,
	}

	r, err := c.ListDetectorRecipes(context.Background(), request)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	fmt.Printf("List: %v", r.Items)
	return
}

```

# Output 
```
test@test-3 % go run test_cloudguard.go
List: []%                                                                                                                        
```

**Comment by bigdatasourav** [comment]
Configuration API is working fine -
```
package main

import (
	"context"
	"fmt"

	"github.com/oracle/oci-go-sdk/v44/cloudguard"
	"github.com/oracle/oci-go-sdk/v44/common"
)

func main() {
	c, err := cloudguard.NewCloudGuardClientWithConfigurationProvider(common.DefaultConfigProvider())
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	tenancyID := "ocid1.tenancy.oc1..*********************"
	request := cloudguard.GetConfigurationRequest{
		CompartmentId: &tenancyID,
	}

	r, err := c.GetConfiguration(context.Background(), request)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	fmt.Printf("List: %v", r.Configuration)
	return
}

``` 
# output 
```
test@test % go run test_cloudguard.go
List: { ReportingRegion=ap-hyderabad-1 Status=ENABLED SelfManageResources=false }% 
```

**Comment by pelliu** [comment]
@bigdatasourav, thanks for letting us know your questions, I looked into the issue, seems like the response of `ListDetectorRecipes`'s body is defined inside of `DetectorRecipeCollection`(https://github.com/oracle/oci-go-sdk/blob/551afdba6f1e966c52ae2c5d78dbef7825a3972a/cloudguard/list_detector_recipes_request_response.go#L95 and https://github.com/oracle/oci-go-sdk/blob/551afdba6f1e966c52ae2c5d78dbef7825a3972a/cloudguard/detector_recipe_collection.go#L20)
for how to use pagination operations(ListXXXXXX), please refer https://github.com/oracle/oci-go-sdk/blob/master/oci.go#L236 and the example https://github.com/oracle/oci-go-sdk/blob/master/example/example_core_pagination_test.go.

**Comment by pelliu** [comment]
@bigdatasourav, just want to follow up - did you try to call r.DetectorRecipeCollection.Items ? does it work for you?

**Comment by bigdatasourav** [comment]
@pelliu Below are the details I have found -

If I set reporting region in ~/.oci/config  -> I am getting values from r.DetectorRecipeCollection.Items and r.Items as well.
If I set any other region in ~/.oci/config  -> No data from r.DetectorRecipeCollection.Items or r.Items

Whether I set reporting region in ~/.oci/config or not --> GetConfiguration API is working fine.

Now my question is except GetConfiguration API, are other cloudguard API's are reporting region dependent?
